hive/src/main/java/com/netease/arctic/hive/catalog/ArcticHiveCatalog.java

```diff
@@ -166,22 +180,30 @@ protected UnkeyedHiveTable loadUnKeyedTable(TableMeta tableMeta) {
         arcticFileIO, tableMetaStore.getConfiguration()), arcticFileIO, tableLocation, client, hiveClientPool);
   }
 
+  public HMSClient getHMSClient() {
+    return hiveClientPool;
+  }
+
   class ArcticHiveTableBuilder extends BaseArcticTableBuilder {
 
     public ArcticHiveTableBuilder(TableIdentifier identifier, Schema schema) {
       super(identifier, schema);
     }
 
+
+
     @Override
     protected void doCreateCheck() {
       super.doCreateCheck();
       try {
-        org.apache.hadoop.hive.metastore.api.Table hiveTable =
-            hiveClientPool.run(client -> client.getTable(
-                identifier.getDatabase(),
-                identifier.getTableName()));
-        if (hiveTable != null) {
-          throw new IllegalArgumentException("Table is already existed in hive meta store:" + identifier);
+        if (!isUpgradeHive) {
```

Comment:
I recommend to add a `allowHiveTableExsited` config for table builder and check existed hive table schema and partition spec.